em-app/src/lib.rs

```diff
@@ -190,33 +216,78 @@ impl ExternalKey for Pk {
     }
 }
 
-fn get_target_report(client: &mut Client, pub_key: &Vec<u8>) -> Result<sgx_isa::Report> {
+fn get_target_report(client: &mut Client, pub_key: &Vec<u8>, config_id: Option<&str>) -> Result<sgx_isa::Report> {
     let result = client.get_target_info().map_err(|e| Error::TargetReport(Box::new(e)))?;
     
     let target_info = Targetinfo::try_copy_from(result.target_info.ok_or(Error::TargetReportInternal("Node Agent returned empty target_info".to_string()))?
                                                                   .from_base64().map_err(|e| Error::TargetReportInternal(format!("Base64 decode failed: {:?}", e)))?
                                                                   .as_ref()
                                                ).ok_or(Error::TargetReportInternal("Failed creating SGX structure from remote target info".to_string()))?;
 
+
     let mut data=[0u8;64];
     Md::hash(Type::Sha256, &pub_key, &mut data).map_err(|e| Error::TargetReportHash(Box::new(e)))?;
+
+    if let Some(id) = config_id {
+        let id = id.from_hex().map_err(|e| Error::ConfigIdIssue(format!("Failed decoding config ID: {}", e)))?;
+        
+        let mut payload=[0u8;65];
+        payload[0] = 1;
+        payload[1..33].copy_from_slice(&data[0..32]);
+        payload[33..65].copy_from_slice(&id[0..32]);
+
+        Md::hash(Type::Sha256, &payload, &mut data[0..32]).map_err(|e| Error::TargetReportHash(Box::new(e)))?;
+        data[32..64].copy_from_slice(&id[0..32]);
```

Comment:
Is there something we can reference that defines the reportdata contents? Or if not, maybe worth a brief explanation here?

Reply:
Added comments here.